src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -5,10 +5,59 @@
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MIN_AGE = 18;
+    private static final int MIN_LOGIN_LENGTH = 6;
+    private static final String INVALID_DATA = "Invalid user data."
+            + " Please check login, password, and age.";
+    private static final String EXISTS_MSG = "User with this login already exists.";
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        validateUser(user);
+        return storageDao.add(user);
+    }
+
+    public void validateUser(User user) {
+        validateNotNull(user);
+        validateLogin(user.getLogin());
+        validatePassword(user.getPassword());
+        validateAge(user.getAge());
+        validateExistingUser(user.getLogin());
+    }
+
+    private void validateNotNull(User user) {
+        if (user == null
+                || user.getLogin() == null
+                || user.getPassword() == null
+                || user.getAge() == null
+        ) {
+            throw new InvalidUserException(INVALID_DATA);
```

Comment:
better use concrete exception messages for each case, not generic one
it'd be ok if you wont use constants for that, and just type messages here
this will lead us to easy exception fixing when they'll be thrown
![image](https://github.com/mate-academy/jv-registration-validation/assets/73041091/7dfc74b1-ee41-4b1b-ae6b-ca4e3e450d6f)